src/Symfony/Component/Process/Process.php

```diff
@@ -465,6 +465,7 @@ public function waitUntil(callable $callback)
                     $this->fallbackStatus['exitcode'] = (int) $data;
                 }
             }
+            usleep(1000);
```

Comment:
wouldn't it make sense to wrap this in `if ($this->isRunning()) {`?

Reply:
BTW, how does this method behave when the callback never returned true but the process is not running anymore?

Reply:
The purpose of `usleep` here is to ease use of CPU. if to wrap it in `if ($this->isRunning())` it would take more operations like reading pipe status, which I guess is not necessary.

Reply:
> BTW, how does this method behave when the callback never returned true but the process is not running anymore?

Just checked it and I got exception after 60 seconds of default timeout.

Reply:
let's break the loop when it's not running instead, isn't it?